File: python/3rd_week/latihanpraktikum3_py/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc:
    # implementasi class pc
    
    # atribut
    procObj = Processor()
    diskObj = Disk()
    ramObj = Ram()
    totalPrice = 0
    
    # constructor
    def __init__(self):
        pass
    
    # setter atribut
    def setProcObj(self, procObj):
        self.procObj = procObj
    
    def setDiskObj(self, diskObj):
        self.diskObj = diskObj
    
    def setRamObj(self, ramObj):
        self.ramObj = ramObj
    
    # getter atribut
    def getProcObj(self):
        return self.procObj
    
    def getDiskObj(self):
        return self.diskObj
    
    def gerRamObj(self):
        return self.ramObj
    
    # method menghitung total harga
    def countTotal(self):
        self.totalPrice = self.procObj.getProcPrice() + self.diskObj.getDiskPrice() + self.ramObj.getRamPrice()
    
    # method print data pc
    def printPc(self):
        self.procObj.printProc()
        self.diskObj.printDisk()
        self.ramObj.printRam()
        print("-Data PC")
        print(f" Harga PC : {self.totalPrice}")
        print("--------------------------------------")

File: php/3rd_week/latihanpraktikum3_php/Pc.php
<?php

include "Processor.php";
include "Disk.php";
include "Ram.php";

class Pc{
    // implementasi class Pc

    // atribut
    private $procObj;
    private $diskObj;
    private $ramObj;
    private $totalPrice = 0;

    //constructor
    function __construct()
    {
    }

    function setProcObj($paramProcObj){
        $this->procObj = $paramProcObj;
    }

    function setDiskObj($paramDiskObj){
        $this->diskObj = $paramDiskObj;
    }

    function setRamObj($paramRamObj){
        $this->ramObj = $paramRamObj;
    }

    function getProcObj(){
        return $this->procObj;
    }

    function getDiskObj(){
        return $this->diskObj;
    }

    function getRamObject(){
        return $this->ramObj;
    }

    function countTotal(){
        $this->totalPrice = $this->procObj->getProcPrice() + $this->diskObj->getDiskPrice() + $this->ramObj->getRamPrice();
    }

    // method print all data
    function printPc(){
        $this->procObj->printProc();
        $this->diskObj->printDisk();
        $this->ramObj->printRam();
        echo "-Data Pc". "<br/>";
        echo " Harga PC : ". $this->totalPrice. "<br/>";
        echo "--------------------------------------------". "<br/>";
    }

    //destructor
    function __destruct()
    {
    }
}
?>